[DataGrid] Variable row height (#3218)
mui-org/material-ui-x: docs/src/pages/components/data-grid/rows/rows.md

{{"demo": "pages/components/data-grid/rows/DenseHeightGrid.js", "bg": "inline"}}

## Variable row height

If you need different rows to have different row heights this can be achieved using the `getRowHeight` prop. The prop is called for each visible row and if the return value is `number` then that `number` will be set as that row's `rowHeight`. If the return value is `null` or `undefined` then the `rowHeight` prop will take effect for the given row.

> {{"demo": "pages/components/data-grid/rows/VariableRowHeightGrid.js", "bg": "inline"}}

Reviewer: This demo could have the toolbar so users can play with the density. Also, the preview doesn't show how to calculate the value based on the factor.

> Changing the `DataGrid` density doesn't affect the rows with variable row height. You can access the `DataGrid` density factor from the `getRowHeight` params.

Reviewer: This could be a warning.

```suggestion
> Changing the `DataGrid` density does not affect the rows with variable row height.
> You can access the density factor from the params provided to the `getRowHeight` prop.
```

{{"demo": "pages/components/data-grid/rows/DenseHeightGrid.js", "bg": "inline"}}

> ## Variable row height

Reviewer: Should it be a sub-section of "Row height"?

```suggestion
### Variable row height
```

> If you need different rows to have different row heights this can be achieved using the `getRowHeight` prop. The prop is called for each visible row and if the return value is `number` then that `number` will be set as that row's `rowHeight`. If the return value is `null` or `undefined` then the `rowHeight` prop will take effect for the given row.

Reviewer 2: Just to avoid repetitions.

```suggestion
If you need some rows to have different row heights this can be achieved using the `getRowHeight` prop. This function is called for each visible row and if the return value is a `number` then that `number` will be set as that row's `rowHeight`. If the return value is `null` or `undefined` then the `rowHeight` prop will take effect for the given row.
```

{{"demo": "pages/components/data-grid/rows/VariableRowHeightGrid.js", "bg": "inline"}}

> Changing the `DataGrid` density does not affect the rows with variable row height.
> You can access the density factor from the params provided to the `getRowHeight` prop

> 

Reviewer: Could you add a warning recommending `getRowHeight` to be be memoized. See https://deploy-preview-3387--material-ui-x.netlify.app/components/data-grid/group-pivot/#master-detail for an example.
